### ISSUE TITLE
NvStrapsReBar.exe displays "UEFI DXE driver status: Unconfigured (0x0000000000000028)" all the time

### ISSUE THREAD

**Issue by fidoboy** [terminatorul/NvStrapsReBar#110]
**System**
 - Motherboard: ASUS P9X79 PRO
 - BIOS Version: 4801 (modded)
 - GPU: nVidia GTX 1660 SUPER
 - [ X] CSM is turned off. **Make sure to confirm this in the BIOS and not with GPU-Z or similar since it can be inaccurate**
 - [ X] 4G decoding is enabled. **Make sure to confirm this in the BIOS and not with GPU-Z or similar since it can be inaccurate**
 - [ X] UEFIPatch is applied (see [Using UEFIPatch](https://github.com/xCuri0/ReBarUEFI/wiki/Using-UEFIPatch) for more information). On some motherboards [DSDT Patching](https://github.com/xCuri0/ReBarUEFI/wiki/DSDT-Patching) is also needed
 - [ X] I have read [Common issues (and fixes)](https://github.com/xCuri0/ReBarUEFI/wiki/Common-issues-(and-fixes))

**Description**

I've managed to boot with 4G enabled in my ASUS P9X79 PRO mainboard with [this patch](https://github.com/xCuri0/ReBarUEFI/blob/master/UEFIPatch/patches.txt) and GPU-Z confirms that it's enabled. I'm not sure if I also need to [use this](https://github.com/xCuri0/ReBarUEFI/wiki/DSDT-Patching). (DSDT Patching)

![GPU-Z](https://github.com/user-attachments/assets/0f66ad1a-d9ee-4909-81e5-a15084d879fd)

Also I've this in my device manager:

![devmanager](https://github.com/user-attachments/assets/ca6609dd-b909-4d1b-ad69-d83c3836f980)

But when execute the NvStrapsReBar.exe it always says "UEFI DXE driver status: Unconfigured (0x0000000000000028)". I've tried my best selecting the P option to set the "Target BAR size" to 256MiB and then S to save the settings. But after reboot my system it says "UEFI DXE driver status: Unconfigured (0x0000000000000028)" again. So no changes and Resizable BAR still disabled. I've tried many times and there is no way to get it working.

What's wrong? Do I need any other UEFI patch than the one referenced before? 

This is the BIOS that I'm using: 
[P9X79PRO.zip](https://github.com/user-attachments/files/17514288/P9X79PRO.zip)

**Comment by fidoboy** [comment]
> Have you tried option O (`Enable override for BAR size mask for PCIe ReBAR capability`)?

No. Should I use it? I didn't know what it was for, so I decided not to touch it.

**Comment by fidoboy** [comment]
> Have you tried option O (`Enable override for BAR size mask for PCIe ReBAR capability`)?

What I did was to uncomment the line "Extend MMIOH limit to fix Above 4G Decoding (X79)" in patches.txt. And with this now my system boots fine with no issues and GPU-Z (I don't know any other procedure to check it) shows that "Above 4G Decode" is enabled. But the issue is that it seems that the NvStrapsReBar.exe has no effect. With every reboot it displays again the same message "UEFI DXE driver status: Unconfigured (0x0000000000000028)"

**Comment by fidoboy** [comment]
I don't know what any other thing should I test or what other info provide here to help. The NvStrapsReBar.exe itself doesn't provide too much info or any clues, so I don't know what's wrong in my setup. However you can take a look at my current BIOS that I've attached into the first message. The NvStrapsReBar.ffs is included as you can confirm by using UEFITool.

**Comment by fidoboy** [comment]
This is what I get when using the W option:

```
DXE Driver configuration:
        isDirty:           0
        OptionFlags:       0x08
        nPciBarSize:       8
        nGPUSelectorCount: 1
                GPUSelector1:  deviceID:            21C4
                GPUSelector1:  subsysVendorID:      FFFF
                GPUSelector1:  subsysDeviceID:      FFFF
                GPUSelector1:  bus:                 FF
                GPUSelector1:  device:              FF
                GPUSelector1:  function:            FF
                GPUSelector1:  barSizeSelector:     2
                GPUSelector1:  overridebarSizeMask: 0

        nGPUConfigCount:   1
                GPUConfig1:    deviceID:        21C4
                GPUConfig1:    subsysVendorID:  1458
                GPUConfig1:    subsysDeviceID:  407D
                GPUConfig1:    bus:             01
                GPUConfig1:    device:          00
                GPUConfig1:    function:        0
                GPUConfig1:    BAR0 base:       0xFA00'0000
                GPUConfig1:    BAR0 top:        0xFAFF'FFFF

        nBridgeCount:      1
                BridgeConfig1: vendorID:        8086
                BridgeConfig1: deviceID:        3C04
                BridgeConfig1: bus:             00
                BridgeConfig1: device:          02
                BridgeConfig1: function:        0
                BridgeConfig1: secondary bus:   01
```

I don't know if it helps.

**Comment by fidoboy** [comment]
I've also noticed that the NVIDIA Control Panel doesn't display any info (yes or no, enable or diable) about "Resizable BAR" into the System Info dialog. I suppose that this is normal because it's not officially supported on this model.

**Comment by fidoboy** [comment]
> Have you tried option O (`Enable override for BAR size mask for PCIe ReBAR capability`)?

Now I've tried it and there is no changes. Using thisssssssssssss option doesn't seems to have any effect on the obtained results. May be that @terminatorul could create a debug version wich provides some additional info to know what's happening in my case (ASUS P9X79).

**Comment by fidoboy** [comment]
Finally solved by installing the 0.3 version of NvStrapsReBar. It seems that the 0.4 release is not working fine.

**Comment by FirstEverTech** [comment]
Hi, can you share your final modded bios for P9X79 Pro?
One more question, what value are you typing in ReBarState? Is it 13 to get 8GB? (2^13 = 8192MB)

![image](https://github.com/user-attachments/assets/b551911a-aeb6-419f-8244-db385abad1bc)

On ASUS Rampage IV Formula you can use value 32 (32: Unlimited BAR size) and set ReBar to 16GB.

![image](https://github.com/user-attachments/assets/8fd3b261-dddd-4d46-869d-d70a4347aba5)

Thanks!
